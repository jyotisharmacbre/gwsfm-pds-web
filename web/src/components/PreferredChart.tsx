import React from 'react';

import { Doughnut, Pie, Polar } from 'react-chartjs-2';
import Grid from '@material-ui/core/Grid';

const data = {
    datasets: [{
      data: [
        11,
        16,
        
      ],
      backgroundColor: [
        '#4cbd7f',
        '#EA8032'
       
      ],
      label: 'My dataset' // for legend
    }],
    labels: [
      'Preferred',
      'Not preferred'
    ]
  };
  const optionsLabels={  
    responsive: true,
    maintainAspectRatio: true,
    legend: {
      display: true,
      position:'bottom'
    }//, maintainAspectRatio:false     
};
export default function PreferredChart(props: {Preferred:number, NotPreferred:number}){
    return (  <Grid item xs={12} sm={12} lg={12} md={12} style={{"textAlign": 'center', "height":"100%"}} id="GridPreferredChart" >
          <Pie  data={data}  options={optionsLabels} 
        /></Grid>
        
             );
   
   
  }
  