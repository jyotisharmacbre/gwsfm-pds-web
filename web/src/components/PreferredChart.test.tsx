import React from 'react';
import ReactDOM from 'react-dom';
import PreferredChart from './PreferredChart';
import renderer from 'react-test-renderer';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

import { Doughnut, Pie, Polar, Bubble, Line } from 'react-chartjs-2';
it('Preferred Chart should render', () => {
    const component = shallow(
        <PreferredChart Preferred={4} NotPreferred={6}  />
    );
   // let tree = component.toJSON();
    expect(component.getElements()).toMatchSnapshot();
});

 it('Graph Grid exist', () => {
   const props ={
    Preferred:4, NotPreferred:6  
   }
     const component = mount(
        <PreferredChart {...props} />
    );

    expect(component.find('#GridPreferredChart').exists);  
 });

 
