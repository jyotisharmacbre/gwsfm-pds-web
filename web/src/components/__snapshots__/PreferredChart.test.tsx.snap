// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should render 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <span
      className="MuiBadge-root"
      onClick={Array []}
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
        focusable="false"
        id="BellIcon"
        name="BellIcon"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
        />
      </svg>
      <span
        className="MuiBadge-badge"
      >
        5
      </span>
    </span>
  </span>
</button>
`;

exports[`Preferred Chart should render 1`] = `
Array [
  <ForwardRef(WithStyles)
    id="GridPreferredChart"
    item={true}
    lg={12}
    md={12}
    sm={12}
    style={
      Object {
        "height": "100%",
        "textAlign": "center",
      }
    }
    xs={12}
  >
    <Pie
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": Array [
                "#4cbd7f",
                "#EA8032",
              ],
              "data": Array [
                11,
                16,
              ],
              "label": "My dataset",
            },
          ],
          "labels": Array [
            "Preferred",
            "Not preferred",
          ],
        }
      }
      options={
        Object {
          "legend": Object {
            "display": true,
            "position": "bottom",
          },
          "maintainAspectRatio": true,
          "responsive": true,
        }
      }
    />
  </ForwardRef(WithStyles)>,
]
`;
